cmd/cosign/cli/verify/verify_attestation.go

```diff
@@ -195,22 +195,6 @@ func (c *VerifyAttestationCommand) Exec(ctx context.Context, images []string) (e
 				return errors.Wrap(err, "unmarshal payload data")
 			}
 
-			predicateURI, ok := options.PredicateTypeMap[c.PredicateType]
-			if !ok {
-				return fmt.Errorf("invalid predicate type: %s", c.PredicateType)
-			}
-
-			// sanity checks
-			if val, ok := payloadData["payloadType"]; ok {
-				// we need to check only given type from the cli flag
-				// so we are skipping other types
-				if predicateURI != val {
-					continue
```

Comment:
I think removing this check altogether causes the `--type` parameter to be completely ignored and the policy to be applied to all attestations found. My reading of the code is that this is not intended. I tried a different take in resolving this issue in the first commit of https://github.com/sigstore/cosign/pull/1672.